cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java

```diff
@@ -342,6 +369,25 @@ private CubeExploreQuery buildCubeSearchQuery(MetricSearchQuery query) {
     return cube.get().findMeasureNames(buildCubeSearchQuery(query));
   }
 
+  @Override
+  public Map<String, MetricProcessorStat> getMetricsProcessorStats() throws Exception {
+    MetricsConsumerMetaTable metaTable = metaTableSupplier.get();
+    Map<String, MetricProcessorStat> processMap = new HashMap<>();
+    for (TopicId topicId : metricsTopics) {
+      MetricsProcessorStats metricProcessorStat = metaTable.getMetricsProcessorStats(new TopicIdMetaKey(topicId));
+      // todo : maybe add messageId to watchdog-api? how to surface messageId in a better way or can we skip it ?
```

Comment:
@anew if we want messageId in nice format, we need to add that to watchdog-api as a class, otherwise currently its in string format, which doesn't look great, should we rather skip messageId in the response ?